### DISCUSSION TITLE
Pasting is not being handled efficiently in Windows

### DISCUSSION THREAD

**Discussion by NewtonChutney** [helix-editor/helix#9243] [General] [answered]
### Summary

When ctrl+v is used to paste in Windows, the insertions happen character by character for some reason..
The same is not true with hx run from wsl..

It appears the text is input as keystrokes and also affects the mode in which helix works! (the pasting action opened the file picker)
And because of this, even if I had been in insert mode earlier.. brackets are being inserted erroneously.. And the formatting is lost too

https://github.com/helix-editor/helix/assets/70827815/6232a41a-9107-4d50-88a3-62bc98602591

### Reproduction Steps

 

1. Launch `hx` and paste from clipboard with ctrl+v (the keybind to paste is configured properly in Windows Terminal)
2. Launch `hx` in a WSL distro and do the same

expected/actual:
The copy functionality should be the same in case it is a terminal emulator issue, but found the issue only on the Windows build

### Helix log

will provide if necessary

### Platform

Windows

### Terminal Emulator

Windows Terminal v1.18.3181.0

### Installation Method

Windows: winget, WSL: Ubuntu apt package

### Helix Version

Ubuntu/WSL and Windows: helix 23.10 (f6021dd0)

Converted to a discussion from: #9242

**Comment by kirawi** [comment, marked as answer]
I've never been able to make bracketed paste work with Windows Terminal even though it's supposed to be supported (might be an upstream issue https://github.com/crossterm-rs/crossterm/issues/737). You have alternatives: `space-p` to paste in the buffer and `C-r` + `*` in prompts. Or you can use an alternative terminal emulator such as Wezterm.

**Reply by NewtonChutney** [reply]
> might be an upstream issue https://github.com/crossterm-rs/crossterm/issues/737)

Ughh.. 😓 Seems like it.. 


>You have alternatives: `space-p` to paste in the buffer and `C-r` + `*` in prompts. Or you can use an alternative terminal emulator such as Wezterm.

Yes, I did look at the `space p` keybind..
In Windows, I'm resorting to using hx on WSL instead for now.. 
I'd have to look into wezterm..

**Reply by NewtonChutney** [reply]
But it's the enterprise environment that is the issue.. When they won't allow WSL nor external app downloads.. ☠️

**Reply by NewtonChutney** [reply]
Doesn't look like wezterm fixes the issue.. Will look into other emulators too..

**Reply by kirawi** [reply]
Could actually be something to do with `conpty` so other emulators may not fix it.

**Reply by NewtonChutney** [reply]
But it does work well when launched from WSL.. In WT that too.. 🤔

**Reply by kirawi** [reply]
You should also create an issue in Windows Terminal for this since it works in WSL but not in native Windows.

**Reply by gabydd** [reply]
From looking at crossterm looks like bracketed paste is not implemented if building for windows even if using a terminal that supports it

**Reply by NewtonChutney** [reply]
Isn't crossterm a Helix dependency.. 
Ahh, so it ain't a terminal emulator issue..

**Reply by kirawi** [reply]
I think it is implemented on Windows though: https://github.com/crossterm-rs/crossterm/blob/08762b3ef4519e7f834453bf91e3fe36f4c63fe7/src/event.rs#L385-L387

`write_ansi` is used for Windows 10+ while the below winapi function is for older versions that don't support bracketed paste.
